sw/device/tests/chip_power_idle_load.c

```diff
@@ -0,0 +1,516 @@
+  dif_alert_handler_class_config_t class_config = {
+      .auto_lock_accumulation_counter =
+          kDifToggleDisabled,  // Whether to automatically lock the accumulation
+                               // counter
+      .accumulator_threshold = UINT16_MAX,  // number of alerts that must fire
+                                            // in order to trigger escalation
+      .irq_deadline_cycles = UINT32_MAX,    // number of cycles this class's
+                                            // associated IRQ handler has,
+      //                                   // to acknowledge the IRQ, before
+      //                                   escalation is triggered
+      .escalation_phases = esc_phases,
+      .escalation_phases_len = ARRAYSIZE(esc_phases),
+      .crashdump_escalation_phase = kDifAlertHandlerClassStatePhase1,
+  };
+  dif_alert_handler_class_config_t class_configs[] = {class_config,
+                                                      class_config};
+  dif_alert_handler_class_t classes[] = {kDifAlertHandlerClassA,
+                                         kDifAlertHandlerClassB};
+
+  // 4. Runtime configuration (struct) for the alert handler:
+  dif_alert_handler_config_t config = {
+      .alerts = alerts,  // list of alerts to configure
+      .alert_classes =
+          alert_classes,  // list of classes to assign each alert to
+      .alerts_len = ARRAYSIZE(
+          alerts),  // lengths of the arrays `alerts` and `alert_classes`
+      .local_alerts = loc_alerts,  // list of local alerts to configure
+      .local_alert_classes =
+          loc_alert_classes,  // list of classes to assign each local alert to
+      .local_alerts_len =
+          ARRAYSIZE(loc_alerts),  // lengths of the arrays `local_alerts` and
+                                  // `local_alert_classes`
+      .classes = classes,         // list of alert classes to configure
+      .class_configs = class_configs,  // list of alert class (escalation
+                                       // protocol) configurations
+      .classes_len = ARRAYSIZE(
+          class_configs),  // length of the arrays `classes` and `class_configs`
+      .ping_timeout =
+          UINT16_MAX,  // ping timeout (16 bits) in cycles - max value to avoid
+                       // timeout and consequent local alert
+  };
+
+  // 5. Configure Alerts
+  // This also locks the configuration.
+  // Once the configuration is locked, it cannot be reconfigured until after a
+  // system reset.
+  alert_handler_testutils_configure_all(
+      &alert_handler, config,
+      kDifToggleEnabled  // "Enable" to lock the configuration
+  );
+
+  // Checks whether alert handler's ping timer is locked
+  bool is_locked;
+  CHECK_DIF_OK(
+      dif_alert_handler_is_ping_timer_locked(&alert_handler, &is_locked));
+  CHECK(is_locked, "Expected alerts to be locked");
+
+  LOG_INFO("SW_INFO: Alert ping is active");
+
+  // ------------------------------------
+  // PWM
+
```

Comment:
similar to the gpio point above, it might be nice to have a pwm monitor on the sv side to make sure this is behaving as expected. But again not critical at the moment.

Reply:
Done,